core/server/worker/src/main/java/alluxio/worker/block/BlockLockManager.java

```diff
@@ -112,7 +112,16 @@ public long lockBlock(long sessionId, long blockId, BlockLockType blockLockType)
       }
       lock = blockLock.writeLock();
     }
-    lock.lock();
+    try {
+      long lockTimeout = Configuration.getInt(PropertyKey.WORKER_BLOCK_LOCK_TIMEOUT);
+      if (!lock.tryLock(lockTimeout, TimeUnit.MILLISECONDS)) {
+        // The socket will be timed out by now, so the client won't receive this exception.
+        throw new RuntimeException(
+            String.format("Failed to acquire block lock after %sms", lockTimeout));
+      }
+    } catch (InterruptedException e) {
+      throw new RuntimeException(e);
+    }
```

Comment:
Could you update the Javadoc? Also, is it fine to do this, since `lockBlock` is used in cases aside from the client's lock block request. Should we instead have another method that takes a timeout?